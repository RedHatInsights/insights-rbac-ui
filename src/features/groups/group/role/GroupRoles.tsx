import React, { Fragment, Suspense, useCallback, useMemo } from 'react';
import { useDispatch } from 'react-redux';
import { Outlet, useParams } from 'react-router-dom';
import Section from '@redhat-cloud-services/frontend-components/Section';
import { Pagination } from '@patternfly/react-core/dist/dynamic/components/Pagination';

// DataView imports
import { DataView, DataViewState } from '@patternfly/react-data-view';
import { DataViewToolbar } from '@patternfly/react-data-view/dist/dynamic/DataViewToolbar';
import { DataViewTable } from '@patternfly/react-data-view/dist/dynamic/DataViewTable';
import { DataViewTextFilter } from '@patternfly/react-data-view';
import DataViewFilters from '@patternfly/react-data-view/dist/cjs/DataViewFilters';
import { SkeletonTableBody, SkeletonTableHead } from '@patternfly/react-component-groups';
import { BulkSelect, BulkSelectValue } from '@patternfly/react-component-groups/dist/dynamic/BulkSelect';

// Component imports
import { GroupRolesEmptyState } from './components/GroupRolesEmptyState';
import { useGroupRoles } from './hooks/useGroupRoles';
import { fetchAddRolesForGroup } from '../../../../redux/groups/actions';
import { getBackRoute } from '../../../../helpers/navigation';
import useAppNavigate from '../../../../hooks/useAppNavigate';
import pathnames from '../../../../utilities/pathnames';
import type { GroupRolesProps } from './types';
import './group-roles.scss';

export const GroupRoles: React.FC<GroupRolesProps> = (props) => {
  const dispatch = useDispatch();
  const navigate = useAppNavigate();
  const { groupId } = useParams<{ groupId: string }>();

  // Use custom hook for ALL business logic
  const {
    roles,
    isLoading,
    filters,
    selection,
    tableRows,
    columns,
    hasActiveFilters,
    hasPermissions,
    isPlatformDefault,
    isAdminDefault,
    pagination,
    fetchData,
    emptyStateProps,
    toolbarButtons,
    group,
  } = useGroupRoles(props);

  // Filter change handler
  const handleFilterChange = useCallback(
    (key: string, newValues: Partial<{ name: string }>) => {
      const newFilters = { ...filters.filters, ...newValues };
      filters.onSetFilters(newFilters);
      fetchData({ name: newFilters.name, offset: 0 });
    },
    [filters, fetchData],
  );

  // Clear all filters
  const clearAllFilters = useCallback(() => {
    filters.onSetFilters({ name: '' });
    fetchData({ offset: 0 });
  }, [filters, fetchData]);

  // Bulk select handler
  const handleBulkSelect = useCallback(
    (value: BulkSelectValue) => {
      if (value === BulkSelectValue.none) {
        selection.onSelect(false);
      } else if (value === BulkSelectValue.page) {
        // Select all items on current page
        selection.onSelect(true, tableRows);
      } else if (value === BulkSelectValue.nonePage) {
        // Deselect all items on current page
        selection.onSelect(false, tableRows);
      }
    },
    [selection, tableRows],
  );

  // Skeleton states
  const loadingHeader = useMemo(() => <SkeletonTableHead columns={columns.map((col) => col.cell)} />, [columns]);

  const loadingBody = useMemo(() => <SkeletonTableBody rowsCount={10} columnsCount={columns.length} />, [columns.length]);

  // Empty state component
  const emptyState = useMemo(() => <GroupRolesEmptyState {...emptyStateProps} />, [emptyStateProps]);

  // Bulk select component
  const bulkSelectComponent = useMemo(() => {
    if (!hasPermissions || isAdminDefault) {
      return undefined;
    }

    const selectedCount = selection.selected?.length || 0;
    const currentPageCount = roles.length; // Items on current page
    const totalCount = pagination.count || 0; // Total items across all pages

    // Calculate if all/some items on current page are selected
    const selectedOnPage = tableRows.filter((row) => selection.selected?.some((sel) => sel.id === row.id)).length;
    const pageSelected = selectedOnPage > 0 && selectedOnPage === currentPageCount;
    const pagePartiallySelected = selectedOnPage > 0 && selectedOnPage < currentPageCount;

    return (
      <BulkSelect
        isDataPaginated={true}
        selectedCount={selectedCount}
        totalCount={totalCount}
        pageCount={currentPageCount}
        pageSelected={pageSelected}
        pagePartiallySelected={pagePartiallySelected}
        onSelect={handleBulkSelect}
      />
    );
  }, [hasPermissions, isAdminDefault, selection.selected, roles.length, pagination.count, tableRows, handleBulkSelect]);

  // Toolbar actions - use the buttons directly from the hook
  const toolbarActions = useMemo(() => {
    if (!hasPermissions || isAdminDefault) {
      return undefined;
    }
    return toolbarButtons;
  }, [hasPermissions, isAdminDefault, toolbarButtons]);

  // Pagination handlers
  const handleSetPage = useCallback(
    (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent, newPage: number) => {
      const offset = (newPage - 1) * pagination.limit;
      fetchData({ offset, limit: pagination.limit });
    },
    [fetchData, pagination.limit],
  );

  const handlePerPageSelect = useCallback(
    (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent, newPerPage: number) => {
      fetchData({ offset: 0, limit: newPerPage });
    },
    [fetchData],
  );

  // Pagination component
  const paginationComponent = useMemo(() => {
    const currentPage = Math.floor(pagination.offset / pagination.limit) + 1;
    return (
      <Pagination
        itemCount={pagination.count || 0}
        perPage={pagination.limit}
        page={currentPage}
        onSetPage={handleSetPage}
        onPerPageSelect={handlePerPageSelect}
        isCompact
      />
    );
  }, [pagination.count, pagination.limit, pagination.offset, handleSetPage, handlePerPageSelect]);

  // Determine active state
  const activeState = isLoading ? DataViewState.loading : roles.length === 0 ? DataViewState.empty : undefined;

  return (
    <Fragment>
      <Section type="content" id="tab-roles">
        <DataView activeState={activeState} selection={hasPermissions && !isAdminDefault ? selection : undefined}>
          <DataViewToolbar
            bulkSelect={bulkSelectComponent}
            pagination={paginationComponent}
            filters={
              <DataViewFilters onChange={handleFilterChange} values={filters.filters}>
                <DataViewTextFilter filterId="name" title="Name" placeholder="Filter by name" />
              </DataViewFilters>
            }
            clearAllFilters={hasActiveFilters ? clearAllFilters : undefined}
            actions={toolbarActions}
          />
          <DataViewTable
            columns={columns}
            rows={tableRows}
            headStates={{ loading: loadingHeader }}
            bodyStates={{
              loading: loadingBody,
              empty: emptyState,
            }}
          />
          <DataViewToolbar pagination={paginationComponent} />
        </DataView>
      </Section>

      {!isPlatformDefault ? (
        <Suspense>
          <Outlet
            context={{
              [pathnames['group-add-roles'].path]: {
                postMethod: (promise: Promise<unknown>) => {
                  navigate(pathnames['group-detail-roles'].link.replace(':groupId', groupId!));
                  if (promise) {
                    promise.then(() => {
                      dispatch(fetchAddRolesForGroup(groupId!, { limit: 20, offset: 0 }));
                      fetchData();
                    });
                  }
                },
              },
              [pathnames['group-roles-edit-group'].path]: {
                group,
                cancelRoute: pathnames['group-detail-roles'].link.replace(':groupId', groupId!),
                submitRoute: pathnames['group-detail-roles'].link.replace(':groupId', groupId!), // Stay on roles tab after edit
              },
              [pathnames['group-roles-remove-group'].path]: {
                postMethod: (promise: Promise<unknown>) => {
                  const backRoute = getBackRoute(pathnames['group-detail-roles'].link.replace(':groupId', groupId!), pagination, {});
                  navigate(backRoute);
                  if (promise) {
                    promise.then(() => {
                      dispatch(fetchAddRolesForGroup(groupId!, { limit: 20, offset: 0 }));
                      fetchData();
                    });
                  }
                },
                // Add cancelRoute so cancel button takes user back to group roles tab
                cancelRoute: pathnames['group-detail-roles'].link.replace(':groupId', groupId!),
                // Add submitRoute for consistent navigation after successful removal
                submitRoute: getBackRoute(pathnames.groups.link, { ...pagination, offset: 0 }, {}),
              },
            }}
          />
        </Suspense>
      ) : null}
    </Fragment>
  );
};

export default GroupRoles;
