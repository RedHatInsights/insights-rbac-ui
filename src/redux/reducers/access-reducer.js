import { GET_PRINCIPAL_ACCESS } from '../../redux/action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const accessInitialState = {
  isLoading: false,
  access: {
    data: [],
    meta: defaultSettings,
  },
};

const setLoadingState = (state) => ({ ...state, isLoading: true });

const setAccess = (state, { payload }) => ({ ...state, access: payload, isLoading: false });

export default {
  [`${GET_PRINCIPAL_ACCESS}_PENDING`]: setLoadingState,
  [`${GET_PRINCIPAL_ACCESS}_FULFILLED`]: setAccess,
};
