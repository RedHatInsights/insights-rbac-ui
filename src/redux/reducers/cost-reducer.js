import { GET_RESOURCE_DEFINITIONS, GET_RESOURCE } from '../../redux/action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const costInitialState = {
  isLoading: false,
  resourceTypes: {
    data: [],
    meta: defaultSettings,
  },
  resources: {},
  loadingResources: 0,
};

const setLoadingState = (state) => ({ ...state, isLoading: true });

const setResourceDefinitions = (state, { payload }) => ({ ...state, resourceTypes: payload, isLoading: false });
const setResource = (state, { payload }) => ({
  ...state,
  resources: { ...state.resources, [payload.links.first.split('/')[5]]: payload.data },
  loadingResources: state.loadingResources - 1,
});

const setResourceLoading = (state) => ({ ...state, loadingResources: state.loadingResources + 1 });

export default {
  [`${GET_RESOURCE_DEFINITIONS}_PENDING`]: setLoadingState,
  [`${GET_RESOURCE_DEFINITIONS}_FULFILLED`]: setResourceDefinitions,
  [`${GET_RESOURCE}_PENDING`]: setResourceLoading,
  [`${GET_RESOURCE}_FULFILLED`]: setResource,
};
