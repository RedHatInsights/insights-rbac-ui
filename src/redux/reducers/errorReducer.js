import { API_ERROR } from '../action-types';

const setErrorState = (_, { payload }) => ({
  errorCode: payload,
});

export default {
  [API_ERROR]: setErrorState,
};
