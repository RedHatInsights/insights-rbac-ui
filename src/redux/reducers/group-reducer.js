import {
  FETCH_GROUP,
  FETCH_SYSTEM_GROUP,
  FETCH_GROUPS,
  RESET_SELECTED_GROUP,
  FETCH_ROLES_FOR_GROUP,
  FETCH_ADD_ROLES_FOR_GROUP,
  FETCH_MEMBERS_FOR_GROUP,
} from '../../redux/action-types';
import omit from 'lodash/omit';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const groupsInitialState = {
  groups: {
    data: [],
    meta: defaultSettings,
  },
  selectedGroup: { addRoles: {}, members: { meta: defaultSettings }, pagination: defaultSettings },
  isLoading: false,
  isRecordLoading: false,
};

const setLoadingState = (state) => ({ ...state, isLoading: true });
const setRecordLoadingState = (state) => ({ ...state, isRecordLoading: true, selectedGroup: { ...state.selectedGroup, loaded: false } });
const setRecordRolesLoadingState = (state) => ({ ...state, isRecordRolesLoading: true, selectedGroup: { ...state.selectedGroup, loaded: false } });
const setGroups = (state, { payload }) => ({ ...state, groups: payload, isLoading: false });
const setSystemGroup = (state, { payload }) => ({ ...state, systemGroup: payload?.data?.filter((group) => group?.platform_default)?.[0] });
const setGroup = (state, { payload }) => ({
  ...state,
  isRecordLoading: false,
  groups: {
    ...state.groups,
    data: state.groups.data.map((group) => ({
      ...group,
      ...(payload.uuid === group.uuid && { ...payload, loaded: true }),
    })),
  },
  selectedGroup: {
    ...state.selectedGroup,
    members: { ...state.selectedGroup.members, data: payload.principals },
    ...omit(payload, ['principals', 'roles']),
    loaded: true,
    pagination: { ...state.selectedGroup.pagination, count: payload.roleCount, offset: 0 },
  },
});
const resetSelectedGroup = (state) => ({ ...state, selectedGroup: undefined });
const setRolesForGroup = (state, { payload }) => ({
  ...state,
  isRecordRolesLoading: false,
  selectedGroup: { ...state.selectedGroup, roles: payload.data, pagination: payload.meta, loaded: true },
});

const setMembersForGroupLoading = (state = {}) => ({
  ...state,
  selectedGroup: {
    ...(state.selectedGroup || {}),
    members: { isLoading: true },
  },
});
const setMembersForGroup = (state, { payload }) => ({
  ...state,
  selectedGroup: {
    ...(state.selectedGroup || {}),
    members: {
      isLoading: false,
      ...payload,
    },
  },
});

const setAddRolesLoading = (state) => ({
  ...state,
  selectedGroup: { ...state.selectedGroup, addRoles: { loaded: false } },
});
const setAddRolesForGroup = (state, { payload }) => ({
  ...state,
  selectedGroup: { ...state.selectedGroup, addRoles: { roles: payload.data, pagination: payload.meta, loaded: true } },
});

export default {
  [`${FETCH_GROUPS}_PENDING`]: setLoadingState,
  [`${FETCH_GROUPS}_FULFILLED`]: setGroups,
  [`${FETCH_SYSTEM_GROUP}_PENDING`]: setLoadingState,
  [`${FETCH_SYSTEM_GROUP}_FULFILLED`]: setSystemGroup,
  [`${FETCH_GROUP}_PENDING`]: setRecordLoadingState,
  [`${FETCH_GROUP}_FULFILLED`]: setGroup,
  [`${FETCH_ROLES_FOR_GROUP}_PENDING`]: setRecordRolesLoadingState,
  [`${FETCH_ROLES_FOR_GROUP}_FULFILLED`]: setRolesForGroup,
  [`${FETCH_MEMBERS_FOR_GROUP}_PENDING`]: setMembersForGroupLoading,
  [`${FETCH_MEMBERS_FOR_GROUP}_FULFILLED`]: setMembersForGroup,
  [`${FETCH_ADD_ROLES_FOR_GROUP}_PENDING`]: setAddRolesLoading,
  [`${FETCH_ADD_ROLES_FOR_GROUP}_FULFILLED`]: setAddRolesForGroup,
  [RESET_SELECTED_GROUP]: resetSelectedGroup,
};
