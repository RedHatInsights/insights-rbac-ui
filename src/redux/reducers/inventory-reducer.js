import { FETCH_INVENTORY_GROUPS, FETCH_INVENTORY_GROUPS_DETAILS } from '../action-types';

export const inventoryGroupsInitialState = {
  isLoading: false,
  resourceTypes: {},
  inventoryGroupsDetails: {},
  total: 0,
};

const setLoadingState = (state) => ({ ...state, isLoading: true });

const setResourceTypes = (state, { payload, meta }) => {
  const data = payload.data.reduce((acc, curr) => ({ ...acc, [curr.name]: { ...curr } }), {});
  return {
    ...state,
    resourceTypes: meta.permissions?.reduce(
      (acc, curr) => ({
        ...acc,
        [curr]: meta?.config?.page > 1 ? { ...(acc?.[curr] || {}), ...data } : data,
      }),
      state.resourceTypes
    ),
    total: payload.meta.count,
    isLoading: false,
  };
};

const setInventoryGroupsDetails = (state, { payload }) => {
  return {
    ...state,
    inventoryGroupsDetails: payload.results.reduce((acc, curr) => ({ ...acc, [curr.id]: curr }), {}),
    isLoading: false,
  };
};

export default {
  [`${FETCH_INVENTORY_GROUPS}_PENDING`]: setLoadingState,
  [`${FETCH_INVENTORY_GROUPS}_FULFILLED`]: setResourceTypes,
  [`${FETCH_INVENTORY_GROUPS_DETAILS}_PENDING`]: setLoadingState,
  [`${FETCH_INVENTORY_GROUPS_DETAILS}_FULFILLED`]: setInventoryGroupsDetails,
};
