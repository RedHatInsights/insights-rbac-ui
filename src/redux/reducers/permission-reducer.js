import { LIST_PERMISSIONS } from '../action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

export const permissionInitialState = {
    isLoading: false,
    permission: {
        data: [],
        meta: { ...defaultSettings, limit: 100 }
    }
};

const setLoadingState = state => ({ ...state, isLoading: true });

const setPermissions = (state, { payload }) => ({ ...state, permission: payload, isLoading: false });

export default {
    [`${LIST_PERMISSIONS}_PENDING`]: setLoadingState,
    [`${LIST_PERMISSIONS}_FULFILLED`]: setPermissions
};
