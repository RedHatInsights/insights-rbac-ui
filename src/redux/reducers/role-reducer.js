import { FETCH_ROLE, FETCH_ROLES, FETCH_ROLE_FOR_USER, FETCH_ROLE_FOR_PRINCIPAL, FETCH_ROLES_FOR_WIZARD } from '../../redux/action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const rolesInitialState = {
  isLoading: false,
  isRecordLoading: false,
  roles: {
    data: [],
    meta: defaultSettings,
  },
  rolesForWizard: {
    data: [],
    meta: defaultSettings,
  },
  selectedRole: {},
};

const setLoadingState = (state) => ({ ...state, isLoading: true });
const setRecordLoadingState = (state) => ({ ...state, isRecordLoading: true });
const setRole = (state, { payload }) => ({ ...state, selectedRole: payload, isRecordLoading: false });
const setRoles = (state, { payload }) => ({ ...state, roles: payload, isLoading: false });
const setRolesWithAccess = (state, { payload }) => ({
  ...state,
  rolesWithAccess: { ...state.rolesWithAccess, [payload.uuid]: payload },
  isRecordLoading: false,
});

const setRolesForWizard = (state, { payload }) => ({ ...state, rolesForWizard: payload, isWizardLoading: false });
const setWizardLoadingState = (state) => ({ ...state, isWizardLoading: true });

export default {
  [`${FETCH_ROLE}_FULFILLED`]: setRole,
  [`${FETCH_ROLE}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLES}_FULFILLED`]: setRoles,
  [`${FETCH_ROLES}_PENDING`]: setLoadingState,
  [`${FETCH_ROLE_FOR_USER}_FULFILLED`]: setRolesWithAccess,
  [`${FETCH_ROLE_FOR_USER}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLE_FOR_PRINCIPAL}_FULFILLED`]: setRolesWithAccess,
  [`${FETCH_ROLE_FOR_PRINCIPAL}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLES_FOR_WIZARD}_FULFILLED`]: setRolesForWizard,
  [`${FETCH_ROLES_FOR_WIZARD}_PENDING`]: setWizardLoadingState,
};
