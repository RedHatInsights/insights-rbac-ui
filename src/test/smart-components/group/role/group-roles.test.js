import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter, Route } from 'react-router-dom';
import { mountToJson } from 'enzyme-to-json';
import promiseMiddleware from 'redux-promise-middleware';
import GroupRoles from '../../../../smart-components/group/role/group-roles';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications';
import { rolesInitialState } from '../../../../redux/reducers/role-reducer';
import { groupsInitialState } from '../../../../redux/reducers/group-reducer';

describe('<GroupPrincipals />', () => {
  let initialProps;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;
  let emptyState;
  let initialState;

  beforeEach(() => {
    initialProps = {};
    mockStore = configureStore(middlewares);
    (emptyState = {
      roleReducer: {
        ...rolesInitialState,
      },
      groupReducer: {
        ...groupsInitialState,
      },
    }),
      (initialState = {
        roleReducer: {
          ...rolesInitialState,
          isLoading: false,
          roles: [],
        },
        groupReducer: {
          ...groupsInitialState,
          isLoading: false,
          groups: {
            identity: {
              user: {
                is_org_admin: true,
              },
            },
          },
          selectedGroup: {
            addRoles: {
              roles: [],
              pagination: {
                count: 1,
                limit: 10,
                offset: 0,
              },
            },
            uuid: '123',
            name: 'Test group',
            description: 'Description',
            platform_default: true,
            roleCount: 11,
            roles: [
              {
                uuid: '123',
                name: 'User role',
                description: 'Description',
                modified: '2020-03-31T19:06:06.682885Z',
                system: true,
                platform_default: true,
              },
            ],
          },
        },
      });
  });

  it('should render empty correctly', () => {
    const store = mockStore(emptyState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/groups/detail/123/roles']} initialIndex={0}>
          <Route path="/groups/detail/:uuid/roles" component={GroupRoles} {...initialProps} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('.pf-c-toolbar button[disabled=false].pf-m-primary')).toHaveLength(0);
    expect(mountToJson(wrapper.find('TableToolbarView'), { mode: 'mount' })).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/groups/detail/123/roles']} initialIndex={0}>
          <Route path="/groups/detail/:uuid/roles" component={GroupRoles} {...initialProps} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('.pf-c-toolbar button[disabled=false].pf-m-primary')).toHaveLength(1);
    expect(mountToJson(wrapper.find('TableToolbarView'), { mode: 'mount' })).toMatchSnapshot();
  });

  it('should fetch group roles on mount', () => {
    const store = mockStore(initialState);
    mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/groups/detail/123/roles']} initialIndex={0}>
          <Route path="/groups/detail/:uuid/roles" component={GroupRoles} {...initialProps} />
        </MemoryRouter>
      </Provider>
    );
    const expectedPayload = [{ type: 'FETCH_ROLES_FOR_GROUP_PENDING' }, { type: 'FETCH_ADD_ROLES_FOR_GROUP_PENDING' }];
    expect(store.getActions()).toEqual(expectedPayload);
  });
});
