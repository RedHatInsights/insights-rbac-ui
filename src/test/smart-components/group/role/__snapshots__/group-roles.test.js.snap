// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupPrincipals /> should render correctly 1`] = `
<ContextProvider
  value={null}
>
  <GroupRoles
    addNotification={[Function]}
    addRoles={[Function]}
    disableAddRoles={true}
    fetchAddRolesForGroup={[Function]}
    fetchRoles={[Function]}
    fetchRolesForGroup={[Function]}
    isChanged={true}
    isLoading={true}
    pagination={
      Object {
        "count": undefined,
        "itemCount": 1,
        "limit": 50,
        "numberOfItems": 50,
        "offset": 0,
      }
    }
    removeRoles={[Function]}
    roles={Array []}
    selectedRoles={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    userIdentity={Object {}}
  />
</ContextProvider>
`;

exports[`<GroupPrincipals /> should render correctly in loading state 1`] = `
<ContextProvider
  value={null}
>
  <GroupRoles
    addNotification={[Function]}
    addRoles={[Function]}
    disableAddRoles={true}
    fetchAddRolesForGroup={[Function]}
    fetchRoles={[Function]}
    fetchRolesForGroup={[Function]}
    isChanged={true}
    isLoading={true}
    pagination={
      Object {
        "count": undefined,
        "itemCount": 1,
        "limit": 50,
        "numberOfItems": 50,
        "offset": 0,
      }
    }
    removeRoles={[Function]}
    roles={Array []}
    selectedRoles={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    userIdentity={Object {}}
  />
</ContextProvider>
`;
